test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -952,6 +849,88 @@ def test_determinism(self):
             for basis_gate in basis_gates:
                 self.assertLessEqual(out.count_ops()[basis_gate], gate_counts[basis_gate])
 
+    @combine(gate=["unitary", "swap"], natural_direction=[True, False])
+    def test_two_qubit_synthesis_to_directional_cx_target(self, gate, natural_direction):
```

Comment:
Could we merge these tests with the target-free versions above by adding e.g. a third argument `via_target=True/False` that then builds a Target or not inside? Then we could reduce the duplication 🙂

Reply:
We could, but not all the tests have an equivalent on the other class, so the unification would take a bit of work. I think that would be out of scope for this PR.